servicenow_connector.py

```diff
@@ -1722,13 +1815,63 @@ def _on_poll(self, param):
             if self._state.get('first_run', True):
                 self._state['first_run'] = False
 
-        if (failed):
+        if failed:
             return action_result.set_status(phantom.APP_ERROR, SERVICENOW_ERR_FAILURES)
 
         self.save_state(self._state)
 
         return action_result.set_status(phantom.APP_SUCCESS)
 
+    def _find_default_severity(self, action_result):
+        try:
+            r = requests.get('{0}rest/severity'.format(self._get_phantom_base_url()), verify=False)  # nosemgrep
+            resp_json = r.json()
+        except Exception as e:
+            return RetVal(action_result.set_status(phantom.APP_ERROR, "Could not get severities \
+                                from platform: {0}".format(e)), None)
+
+        if r.status_code == 401:
+            return RetVal(action_result.set_status(phantom.APP_ERROR, "Could not get severities \
+                from platform: {0}".format(resp_json.get('message', 'Authentication Error'))), None)
+
+        if r.status_code != 200:
+            return RetVal(action_result.set_status(phantom.APP_ERROR, "Could not get severities \
+                from platform: {0}".format(resp_json.get('message', 'Unknown Error'))), None)
+
+        severity = None
+
+        for severity_data in resp_json['data']:
+            if severity_data.get('is_default', False):
+                severity = severity_data['name']
+                break
+
+        return RetVal(phantom.APP_SUCCESS, severity)
+
+    def _validate_custom_severity(self, action_result, severity):
+
+        try:
+            r = requests.get('{0}rest/severity'.format(self._get_phantom_base_url()), verify=False)  # nosemgrep
+            resp_json = r.json()
+        except Exception as e:
+            return RetVal(action_result.set_status(phantom.APP_ERROR, "Could not get severities \
+                            from platform: {0}".format(e)), None)
+
+        if r.status_code == 401:
+            return RetVal(action_result.set_status(phantom.APP_ERROR, "Could not get severities \
+                from platform: {0}".format(resp_json.get('message', 'Authentication Error'))), None)
+
+        if r.status_code != 200:
+            return RetVal(action_result.set_status(phantom.APP_ERROR, "Could not get severities \
+                from platform: {0}".format(resp_json.get('message', 'Unknown Error'))), None)
+
+        severities = [s['name'] for s in resp_json['data']]
+
+        if severity not in severities:
+            return RetVal(action_result.set_status(phantom.APP_ERROR, "Supplied severity, {0}, \
+                not found in configured severities: {1}".format(severity, ', '.join(severities))), None)
+        else:
+            return RetVal(phantom.APP_SUCCESS, {})
```

Comment:
The `else` block is not needed here, right?

Reply:
Yes. I have removed the else block.